templates/application_minio.yaml

```diff
@@ -0,0 +1,42 @@
+{{- if .Values.minio.enable }}
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: minio-operator
+  namespace: argocd
+  annotations:
+    argocd.argoproj.io/sync-wave: "2"
+  finalizers:
+  - resources-finalizer.argocd.argoproj.io
+spec:
+  destination:
+    server: 'https://kubernetes.default.svc'
+    namespace: minio-operator
+  project: default
+  source:
+    repoURL: 'https://operator.min.io'
+    chart: operator
+    targetRevision: 5.0.9
+    helm:
+      releaseName: minio-operator
+      values: |
+        operator:
+          env:
+          - name: MINIO_SERVER_URL
+            value: http://localhost
+          - name: MINIO_BROWSER_REDIRECT_URL
+            value: "http://localhost/minio/"
+        console:
+          ingress:
+            enabled: {{ .Values.minio.ingress.enable }}
+            annotations:
+              nginx.ingress.kubernetes.io/rewrite-target: /minio/$1
```

Comment:
cannot use rewrite target as it needs to be either at root or subdomain: https://github.com/minio/minio/discussions/18054#discussioncomment-7040342